coap-gateway/service/resourceDirectory.go

```diff
@@ -110,7 +163,18 @@ func resourceDirectoryPublishHandler(req *mux.Message, client *Client) {
 		client.logAndWriteErrorResponse(fmt.Errorf("unable to get encoder for accepted type %v requested by device %v: %w", accept, authCtx.GetDeviceID(), err), coapCodes.InternalServerError, req.Token)
 		return
 	}
-	out, err := encode(w)
+
+	type wkRdResponse struct {
+		Links schema.ResourceLinks `json:"links,omitempty"`
+	}
+	var resp wkRdResponse
+	resp.Links = make(schema.ResourceLinks, len(w.Links))
+	for i, link := range w.Links {
+		resp.Links[i].Href = link.Href
+		resp.Links[i].InstanceID = link.InstanceID
```

Comment:
why you don't send deviceID and anchor...
![image](https://user-images.githubusercontent.com/23374974/144106872-91599efb-249b-4ed0-83c0-51871f0226ef.png)

Reply:
Reverted to original after SM discussion.